pipeline/scripts/13-excelExport.R

```diff
@@ -264,143 +267,217 @@ IS_neg$Intensity <- as.numeric(as.character(IS_neg$Intensity))
-  labs(x='',y='Intensity') +
-  theme(axis.text.x=element_text(angle = 90, hjust = 1, vjust = 0.5, size=8))
+  labs(x='',y='Intensity')
 
-IS_pos_plot <- ggplot(IS_pos, aes(Sample,Intensity)) +
+IS_pos_line_plot <- ggplot(IS_pos, aes(Sample_level, Intensity)) +
   ggtitle("Interne Standaard (Pos)") +
   geom_point(aes(col = HMDB.name)) +
   geom_line(aes(col = HMDB.name, group = HMDB.name)) +
-  labs(x = '', y = 'Intensity') +
-  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5, size = 8))
+  labs(x = '', y = 'Intensity')
 
-IS_sum_plot <- ggplot(IS_summed, aes(Sample, Intensity)) +
+IS_sum_line_plot <- ggplot(IS_summed, aes(Sample_level, Intensity)) +
   ggtitle("Interne Standaard (Sum)") +
   geom_point(aes(col = HMDB.name)) +
   geom_line(aes(col = HMDB.name, group = HMDB.name)) +
-  labs(x = '', y = 'Intensity') +
-  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5, size = 8))
-
-w <- 8 + 0.2 * len
-ggsave(paste0(outdir,"/plots/IS_line_all_neg.png"), plot = IS_neg_plot, height = w/2.5, width = w, units = "in")
-ggsave(paste0(outdir,"/plots/IS_line_all_pos.png"), plot = IS_pos_plot, height = w/2.5, width = w, units = "in")
-ggsave(paste0(outdir,"/plots/IS_line_all_sum.png"), plot = IS_sum_plot, height = w/2.5, width = w, units = "in")
-
-
-# Barplot voor selectie aan interne standaarden voor alle data
-IS_sum_selection <- c('2H8-Valine (IS)', '2H3-Leucine (IS)', '2H3-Glutamate (IS)', '2H4_13C5-Arginine (IS)', '13C6-Tyrosine (IS)')
-IS_pos_selection <- c('2H4-Alanine (IS)', '13C6-Phenylalanine (IS)', '2H4_13C5-Arginine (IS)', '2H3-Propionylcarnitine (IS)', '2H9-Isovalerylcarnitine (IS)')
+  labs(x = '', y = 'Intensity')
+
+# add theme to ggplot functions
+IS_sum_line_plot <- theme_IS_line(IS_sum_line_plot)
+IS_neg_line_plot <- theme_IS_line(IS_neg_line_plot)
+IS_pos_line_plot <- theme_IS_line(IS_pos_line_plot)
+
+# save plots to disk
+w <- 8 + 0.2 * sample_count
+ggsave(paste0(outdir,"/plots/IS_line_all_neg.png"), plot = IS_neg_line_plot, height = w/2.5, width = w, units = "in")
+ggsave(paste0(outdir,"/plots/IS_line_all_pos.png"), plot = IS_pos_line_plot, height = w/2.5, width = w, units = "in")
+ggsave(paste0(outdir,"/plots/IS_line_all_sum.png"), plot = IS_sum_line_plot, height = w/2.5, width = w, units = "in")
+
+##########
+##### bar plots with a selection of IS
+##########
 IS_neg_selection <- c('2H2-Ornithine (IS)', '2H3-Glutamate (IS)', '2H2-Citrulline (IS)', '2H4_13C5-Arginine (IS)', '13C6-Tyrosine (IS)')
+IS_pos_selection <- c('2H4-Alanine (IS)', '13C6-Phenylalanine (IS)', '2H4_13C5-Arginine (IS)', '2H3-Propionylcarnitine (IS)', '2H9-Isovalerylcarnitine (IS)')
+IS_sum_selection <- c('2H8-Valine (IS)', '2H3-Leucine (IS)', '2H3-Glutamate (IS)', '2H4_13C5-Arginine (IS)', '13C6-Tyrosine (IS)')
 
-IS_neg_selection_barplot <- ggplot(subset(IS_neg, HMDB.name %in% IS_neg_selection), aes(Sample,Intensity)) +
+# add minimal intensity lines based on matrix (DBS or Plasma) and machine mode (neg, pos, sum)
```

Comment:
Can matrix be something else than DBS or Plasma?  In that case, the script will probably crash as there is not else statement.

Reply:
yes, this is what commit 1338c87 was about (see description pull request) and this is validated in the validation report